<div class="container">

<table style="width: 100%;"><tr>
<td>haploCrossMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Haplotype crossing matrix generation</h2>

<h3>Description</h3>

<p>Generation of the haplotype crossing matrix associated to a <code>Genome</code>
object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">haploCrossMatrix(genomeObj)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>genomeObj</code></td>
<td>
<p>a <code>Genome</code> object</p>
</td>
</tr></table>
<h3>Details</h3>

<p>A crossover matrix is a square matrix of size equal to the number of
haplotypes. It describes for each combination of two gametic haplotypes
the genotype index resulting from their syngamy. In the general case it
is not a symmetrical matrix (it is if a single haploid locus with a single
allele is defined), because the transmission of haploid loci is only
maternal, therefore non-symmetrical as is the transmission of diploid
loci. It is therefore necessary to enter the haplotype frequencies of
male gametes in the columns and the haplotype frequencies of female
gametes in the rows during the calculations (this is done in the
simulations).
</p>


<h3>Value</h3>

<p>An haplotype crossing matrix.
</p>


<h3>Author(s)</h3>

<p>Ehouarn Le Faou
</p>


</div>