<div class="container">

<table style="width: 100%;"><tr>
<td>setPopulation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Setting a population</h2>

<h3>Description</h3>

<p>Generation of a population by providing all the necessary ingredients for
its definition, including a genome, a mutation matrix and a selection regime.
</p>


<h3>Usage</h3>

<pre><code class="language-R">setPopulation(
  name,
  size,
  dioecy,
  genomeObj,
  mutMatrixObj,
  selectionObj,
  selfRate = 0,
  demography = F,
  growthRate = 0,
  initPopSize = NULL,
  initGenoFreq = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>the name of the population</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>the population size</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dioecy</code></td>
<td>
<p>logical indicating whether the simulated population is
dioecious or hermaphroditic</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genomeObj</code></td>
<td>
<p>a <code>Genome</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutMatrixObj</code></td>
<td>
<p>a <code>MutationMatrix</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selectionObj</code></td>
<td>
<p>a <code>Selection</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selfRate</code></td>
<td>
<p>the selfing rate</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>demography</code></td>
<td>
<p>a logic indicating whether the population should have
a demography (stochasticity in the number of individuals present in the
population + logistic growth with carrying capacity equal to the <code>size</code>
parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>growthRate</code></td>
<td>
<p>a <code>Genome</code> object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initPopSize</code></td>
<td>
<p>the initial size of the population. It is necessarily
equal to <code>size</code> if the population has no <code>demography</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initGenoFreq</code></td>
<td>
<p>a vector of the size of the genotype number
describing the initial allele frequencies common to all simulations</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>A population is defined strictly by a name, a size, a sexual system
(dioecy or hermaphodite), and the three objects defined previously:
genome, mutation matrix and selection. In addition to that, it is
possible to define
- a selfing rate (by default equal to 0)
- a vector of initial genotypic frequencies
- a demography
</p>
<p>Two demographic regimes are possible: no demography, i.e. a fixed population
size, or demography, i.e. a population where the size fluctuates
stochastically. The boolean argument 'demography' is used to define whether
there should be stochasticity. For a fixed population size, it is therefore
sufficient to define that 'demography = FALSE' (default) and to set the
desired population size with the 'popSize' parameter.
</p>
<p>For a fluctuating demography, 'demography' must be 'TRUE' and three other
parameters are then needed: the initial population size ('initPopSize'),
the population growth rate ('growthRate') and the carrying capacity of the
population (the population size, 'popSize').
</p>
<p>It is also possible to avoid defining a population size altogether, by
setting off the genetic drift ('drift' parameter). This will allow the
model to be simulated deterministically.
</p>


<h3>Value</h3>

<p>a <code>Population</code> object
</p>


<h3>Author(s)</h3>

<p>Ehouarn Le Faou
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Definition of a population in its simplest form:
DL &lt;- list(dl = c("A", "a"))
HL &lt;- list(hl = c("B", "b"))
mutations &lt;- list(
  mutation(from = "A", to = "a", rate = 1e-3),
  mutation(from = "B", to = "b", rate = 1e-3)
)
genomeObj &lt;- setGenome(listHapLoci = HL, listDipLoci = DL)
pop &lt;- setPopulation(
  name = "A",
  size = 1000,
  dioecy = TRUE,
  genomeObj = genomeObj,
  selectionObj = setSelectNeutral(genomeObj),
  mutMatrixObj = setMutationMatrix(genomeObj, mutations = mutations)
)

</code></pre>


</div>