<div class="container">

<table style="width: 100%;"><tr>
<td>easyManhattan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interactive Manhattan plot labels</h2>

<h3>Description</h3>

<p>Interactive labelling of Manhattan plots using 'shiny' and 'plotly'
interface.
</p>


<h3>Usage</h3>

<pre><code class="language-R">easyManhattan(
  data,
  chrom = "chrom",
  pos = "pos",
  p = "p",
  labs = "rsid",
  startLabels = NULL,
  pcutoff = 5e-08,
  chromGap = NULL,
  chromCols = c("royalblue", "skyblue"),
  sigCol = "red",
  alpha = 0.7,
  labelDir = "horiz",
  xlab = "Chromosome position",
  ylab = expression("-log"[10] ~ "P"),
  xlim = NULL,
  ylim = NULL,
  outline_col = NA,
  shapeScheme = 16,
  size = 6,
  width = ifelse(transpose, 600, 1000),
  height = ifelse(transpose, 800, 600),
  lineLength = 60,
  npoints = max(c(nrow(data)/5, 1e+06)),
  nplotly = 1e+05,
  npeaks = NULL,
  span = 2e+07,
  transpose = FALSE,
  filename = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>The dataset (data.frame or data.table) for the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrom</code></td>
<td>
<p>The column of chomosome values in <code>data</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>The column of SNP positions in <code>data</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>The column of p values in <code>data</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labs</code></td>
<td>
<p>The column of labels in <code>data</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>startLabels</code></td>
<td>
<p>Vector of initial labels. With a character vector, labels
are identified in the column specified by <code>labs</code>. With a numeric vector,
points to be labelled are referred to by row number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to 5E-08.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromGap</code></td>
<td>
<p>Size of gap between chromosomes along the x axis in base
pairs. If <code>NULL</code> this is automatically calculated dependent on the size of
the genome. Default is around 3E07 for a human genome, and smaller for
smaller genomes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromCols</code></td>
<td>
<p>A vector of colours for points by chromosome. Colours are
recycled dependent on the length of the vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigCol</code></td>
<td>
<p>Colour for statistically significant points. Ignored if set to
<code>NA</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Transparency for points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelDir</code></td>
<td>
<p>Option for label lines. See <code>easylabel()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>x axis title. Accepts expressions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>y axis title. Accepts expressions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlim</code></td>
<td>
<p>The x limits (x1, x2) of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylim</code></td>
<td>
<p>The y limits of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outline_col</code></td>
<td>
<p>Colour of symbol outlines. Passed to <code>easylabel()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shapeScheme</code></td>
<td>
<p>A single symbol for points or a vector of symbols. Passed
to <code>easylabel()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>Specifies point size. Passed to <code>easylabel()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p>Width of the plot in pixels. Saving to pdf scales 100 pixels to
1 inch.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>height</code></td>
<td>
<p>Height of the plot in pixels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lineLength</code></td>
<td>
<p>Initial length of label lines in pixels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>npoints</code></td>
<td>
<p>Maximum number of points to plot when saving the final plot to
pdf. By default plots with &gt;1 million points are thinned to speed up
plotting. Setting a value of <code>NA</code> will plot all points.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nplotly</code></td>
<td>
<p>Maximum number of points to display via plotly. We recommend
the default setting of 100,000 points (or fewer).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>npeaks</code></td>
<td>
<p>Number of peaks to label initially.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>span</code></td>
<td>
<p>a peak is defined as the most significant SNP within a window of
width span centred at that SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transpose</code></td>
<td>
<p>Logical whether to transpose the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Filename for saving to pdf.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments passed to <code>easylabel()</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>By default no return value. If <code>output_shiny = FALSE</code> or the shiny
button 'Export plotly &amp; exit' is pressed, a plotly figure is returned.
See <code>easylabel()</code>.
</p>


<h3>See Also</h3>

<p><code>easylabel()</code> <code>easyVolcano()</code>
</p>


</div>