<div class="container">

<table style="width: 100%;"><tr>
<td>EBglmnet-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Empirical Bayesian Lasso (EBlasso) and Elastic Net (EBEN) Methods for Generalized Linear Models</h2>

<h3>Description</h3>

<p> Fast Empirical Bayesian Lasso (EBlasso) and Elastic Net (EBEN) are generalized linear regression methods for variable selections and effect estimations. 
Similar as <code>lasso</code> and <code>elastic net</code> implemented in the package <span class="pkg">glmnet</span>, <span class="pkg">EBglmnet</span> features 
the capabilities of handling <code class="reqn">p&gt;&gt;n</code> data, where <code>p</code> is the number of variables and <code>n</code> is 
the number of samples in the regression model, and inferring a sparse solution such that irrelevant variables 
will have exactly zero value on their regression coefficients. Additionally, there are several unique features in <span class="pkg">EBglmnet</span>: <br><br>
1) Both <code>EBlasso</code> and <code>EBEN</code> can select more than <code>n</code> nonzero effects. <br>
2) EBglmnet also performs hypothesis testing for the significance of nonzero estimates. <br><br>
There are three sets of hierarchical prior distributions implemented in <span class="pkg">EBglmnet</span>: <br><br>
1) EBlasso-NE is a two-level prior with (normal + exponential) distributions for the regression coefficients.<br>
2) EBlasso-NEG is a three-level hierarchical prior with (normal + exponential + gamma) distributions.<br>
3) EBEN implements a normal and generalized gamma hierarchical prior.<br></p>
<p>While those sets of priors are all "peak zero and flat tails", <code>EBlasso-NE</code> assigns more probability mass to the tails, resulting in more nonzero estimates having large <code class="reqn">p</code>-values. In contrast, <code>EBlasso-NEG</code> has a third level constraint on the <code>lasso</code> prior, which results in higher probability mass around zero, thus more sparse results in the final outcome. Meanwhile, <code>EBEN</code> encourages a grouping effect such that highly correlated variables can be selected as a group.
Similar as the relationship between <code>elastic net</code> and <code>lasso</code>, there are two parameters <code class="reqn">(\alpha, \lambda)</code> required for <code>EBEN</code>, and it is reduced to <code>EBlasso-NE</code> when parameter <code class="reqn">\alpha = 1</code>. We recommend using EBlasso-NEG when there are a large number of candidate effects, using EBlasso-NE when effect sizes are relatively small, and using EBEN when groups of highly correlated variables such as co-regulated gene expressions are of interest.
<br><br> 
Two models are available for both methods: linear regression model and logistic regression model. Other features in this package includes:<br> 
* 1 *     epistasis (two-way interactions) can be included for all models/priors; <br>
* 2 *     model implemented with memory efficient <code>C</code> code; <br>
* 3 *     LAPACK/BLAS are used for most linear algebra computations. <br><br></p>
<p>Several simulation and real data analysis in the reference papers demonstrated that <span class="pkg">EBglmnet</span> enjoys better performance than <code>lasso</code> and <code>elastic net</code> methods in terms of power of detection, 
false discover rate, as well as encouraging grouping effect when applicable. <br></p>
<p>Key Algorithms are described in the following paper: <br>
1. EBlasso-NEG: (Cai X., Huang A., and Xu S., 2011), (Huang A., Xu S., and Cai X., 2013) <br>
2. EBlasso-NE: (Huang A., Xu S., and Cai X., 2013) <br>
3. group EBlasso: (Huang A., Martin E., et al. 2014) <br>
4. EBEN: (Huang A., Xu S., and Cai X., 2015)<br>
5. Whole-genome QTL mapping: (Huang A., Xu S., and Cai X., 2014) <br></p>
<p>EBglmnet version after V5 will not support the following. For those functionalities, please refer to the 'cran' package 'EBEN'. 
- Two way interaction (epistasis) will not be supported;
- Group EBlasso will not be supported.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> EBglmnet</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 6.0</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2016-01-15</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> gpl</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Anhui Huang, Dianting Liu <br>
Maintainer: Anhui Huang &lt;anhuihuang@gmail.com&gt;
</p>


<h3>References</h3>

<p>Huang, A., Xu, S., and Cai, X. (2015). Empirical Bayesian elastic net for multiple quantitative trait locus mapping. Heredity 114(1): 107-115.<br><br>	
Huang, A., E. Martin, et al. (2014). "Detecting genetic interactions in pathway-based genome-wide association studies." Genet Epidemiol 38(4): 300-309.<br><br>
Huang, A., S. Xu, et al. (2014). "Whole-genome quantitative trait locus mapping reveals major role of epistasis on yield of rice." PLoS ONE 9(1): e87330.<br><br>
Huang, A. (2014). "Sparse model learning for inferring genotype and phenotype associations." Ph.D Dissertation. University of Miami(1186).<br><br>	
Huang A, Xu S, Cai X. (2013). Empirical Bayesian LASSO-logistic regression for multiple binary trait locus mapping. BMC genetics  14(1):5. <br><br>
Cai, X., Huang, A., and Xu, S. (2011). Fast empirical Bayesian LASSO for multiple quantitative trait locus mapping. BMC Bioinformatics 12, 211.<br><br></p>


</div>