<div class="container">

<table style="width: 100%;"><tr>
<td>EBPRS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Main function</h2>

<h3>Description</h3>

<p>Clean the dataset, extract information from raw data and calculate effect sizes.
(Please notice that there are some requirements for the training and testing datasets.)
</p>


<h3>Usage</h3>

<pre><code class="language-R">EBPRS(train, test, N1, N0, robust = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>train</code></td>
<td>
<p>training dataset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p>testing dataset (list) including fam, bed, bim, which can be generated from function read_plink in our package. If missing(test)=T, the function will use all SNPs in training dataset by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N1</code></td>
<td>
<p>case number</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N0</code></td>
<td>
<p>control number</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>robust</code></td>
<td>
<p>T/F, indicator that whether robust estimation is needed. The function will run faster when robust is set to F. The default is T.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The raw training data should be a
data.fame including A1, A2, OR, P, SNP (NOTE that the colnames should be exactly consistent with the above).
</p>
<p>The SNP column (rsid) is used for indexing.
</p>
<p>An example training dataset can be acquired using data("traindat")
</p>
<p>"test" file can be generated from read_plink("path_to_test_plink_bfile")
</p>
<p>test is a list, which has test$fam (6 columns with information on samples), test$bim (6 columns with information on SNPs), test$bed (genotypes matrix 0, 1, 2)
</p>
<p>Note that in real data, we usually use beta0 = m/20 as the default setting for the EM algorithm,
which is accurate enough in most cases and will have little influence on the prediction performance.
If more accurate parameter estimation is required, we provide a robust estimation (by setting robust=T),
integrating our data-driven bootstrap-based parameter tuning method. This can
derive the best parameter for robust estimation, while more time is needed.
</p>


<h3>Value</h3>

<p>A list containing
data.frame (result): combining the summary statistics and estimated effect sizes (eff)
</p>
<p>estimated effect sizes (eff)
</p>
<p>estimated mu (muHat)
</p>
<p>estimated sigma2 (sigmaHat2)
</p>
<p>estimated proportion of non-associated SNPs (pi0)
</p>
<p>estimated variance of effect sizes of associated SNPs (sigma02)
</p>
<p>If the test file is provided the results also include:
</p>
<p>predictive r2 (r2)
</p>
<p>AUC (AUC)
</p>
<p>estimated polygenic risk score (S)
</p>


<h3>Author(s)</h3>

<p>Shuang Song, Wei Jiang, Lin Hou and Hongyu Zhao
</p>


<h3>References</h3>

<p>Song S, Jiang W, Hou L, Zhao H (2020) Leveraging effect size distributions to improve polygenic risk scores derived from summary statistics of genome-wide association studies. PLoS Comput Biol 16(2): e1007565. https://doi.org/10.1371/journal.pcbi.1007565
</p>


</div>