<div class="container">

<table style="width: 100%;"><tr>
<td>DER</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DIFFERENTIATING SAMPLES USING RARITY, HETEROGENEITY, EVENNESS, TAXONOMIC, PHYLOGENETIC AND FUNCTIONAL DIVERSITY INDICES</h2>

<h3>Description</h3>

<p>An algorithm for differentiating samples on the basis of the rarity, heterogeneity, evenness, taxonomic/phylogenetic and functional diversity indices that better reflect the differences among assemblages.</p>


<h3>Usage</h3>

<pre><code class="language-R">DER(data, Samples, Species, Taxon, TaxonFunc=NULL, TaxonPhyl=NULL,
pos=NULL, varSize="Richness", varColor="Rarity.G", Index=NULL, corr="sqrt",
palette= "heat.colors", size=c(1,5),digitsS=1, digitsC=2, ncolor=100,
transparency=1, references=TRUE, a=1.5, q=2.5, ResetPAR=TRUE, PAR=NULL,
dbFD=NULL, LEGENDS=NULL, TEXT=NULL, COLOR=c("#EEC591FF", "black", "grey50"),
file1="Diversity indices.csv", file2="Polar coordinates.csv",
file3="Indices and area of the polygon.csv", na="NA", dec=",", row.names=FALSE, save=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Data file with the taxonomy, abundance of the species and functional traits (optional). The format of the file must be: an optional column with the position of labels' samples in the DER plot (blue column) in the same order as the variables with the species' abundance in the samples (red columns), the columns with the taxonomy of the species (as many as needed, green columns), the columns with the abundance of the species in each sample (red columns) and optionally the columns with the functional traits of the species. Each row is an unique species, genus, family, etc.
</p>

<table><tr>
<td style="text-align: right;">
<img src="../help/figures/EcoIndRF1.jpg" width="50" alt="EcoIndRF1.jpg">
</td>
</tr></table>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Samples</code></td>
<td>
<p>Variables with the abundance of the species in each sample: sampling sites, dates, etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Species</code></td>
<td>
<p>Variable with the name of the species (without including the genus). It may be other node of the phylogenetic tree, such as the genus, family, etc., for genus level phylogenies, family level phylogenies, etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Taxon</code></td>
<td>
<p>Variables with the taxonomy of the species (taxonomic diversity), as many levels as needed but without including the variable with the node of the argument <em>Species</em>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TaxonFunc</code></td>
<td>
<p>Optionally variables with the functional traits (functional diversity).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TaxonPhyl</code></td>
<td>
<p>Optionally the name of the RData file of the class phylo with the phylogeny. The file must be in the working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>Optionally it is possible to indicate a column with the position of labels' samples in the DER plot. It must be as many as the number of samples and in the same order than the variables described in the argument <em>Samples</em>. Values of 1, 2, 3 and 4, respectively indicate positions below, to the left of, above and to the right of the specified coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varSize</code></td>
<td>
<p>This variable defines the size of the bubble in the DER plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>varColor</code></td>
<td>
<p>This variable defines the color gradient of the bubbles in the DER plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Index</code></td>
<td>
<p>The four/five indices used in the DER algorithm. If it is NULL the algorithm select one, index of rarity, one of heterogeneity, one of evenness one of taxonomy and one of the functional group (if functional traits are provided in the argument <em>TaxonFunc</em>) that achieve a higher dispersion among samples in a polar coordinates system.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corr</code></td>
<td>
<p>Character string specifying the correction method to use, in the function dbFD, when the species-by-species distance matrix cannot be represented in a Euclidean space. Options are "sqrt" (default), "cailliez", "lingoes" or "none".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palette</code></td>
<td>
<p>The color gradient of the bubbles may be one of these palettes: "heat.colors", "terrain.colors", "gray.colors", "topo.colors" or "cm.colors", or any other option defined by the user.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>Range of size of the bubbles. Two values: minimum and maximum size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digitsS</code></td>
<td>
<p>Number of digits of the bubble size legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>digitsC</code></td>
<td>
<p>Number of digits of the color legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncolor</code></td>
<td>
<p>Gradient color of the color legend.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>transparency</code></td>
<td>
<p>Transparency of the color gradient, from 0 to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>references</code></td>
<td>
<p>If it is TRUE the reference points are depicted on the DER plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p>Scale of Renyi diversity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>Scale of Tsallis diversity.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ResetPAR</code></td>
<td>
<p>If it is FALSE, the default condition of the function PAR of the package StatR is not placed and maintained those defined by the user in previous graphics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PAR</code></td>
<td>
<p>It accesses the function PAR of the package StatR that allows to modify many different aspects of the graph.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dbFD</code></td>
<td>
<p>It accesses the function dbFD which allows to specify the arguments that calculates the functional diversity indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LEGENDS</code></td>
<td>
<p>It allows to modify the legend of the bubble size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>TEXT</code></td>
<td>
<p>It allows to modify the text of the labels in the bubbles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>COLOR</code></td>
<td>
<p>A vector with three values: color of the ellipse, color of the points in the legend of the size of the bubbles and color of the references points in the ellipse, respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file1</code></td>
<td>
<p>CSV FILES. Filename with values of total abundance, richness and the rarity, heterogeneity, evenness, taxonomic, phylogenetic and functional diversity indices of each sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file2</code></td>
<td>
<p>CSV FILES. Filename with the polar coordinates of all samples considering the four/five selected indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file3</code></td>
<td>
<p>CSV FILES. Filename with the area of the convex hull (alpha=6) and Euclidean distance obtained in the polar coordinates system for all combinations of the indices.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na</code></td>
<td>
<p>CSV FILE. Text that is used in the cells without data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dec</code></td>
<td>
<p>CSV FILE. It defines if the comma "," is used as decimal separator or the dot ".".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>row.names</code></td>
<td>
<p>CSV FILE. Logical value that defines if identifiers are put in rows or a vector with a text for each of the rows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save</code></td>
<td>
<p>If it is TRUE, the CSV files are saved.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><strong>DER algorithm</strong>
</p>
<p>The steps of DER algorithm are described below:
</p>
<p><strong>1.</strong> The function DER calculates the most often used indices (see below): a total of 39 indices that includes 2 of rarity, 14 of heterogeneity, 7 of evenness, 2 of taxonomic diversity, 8 of phylogenetic diversity and 6 of functional diversity. It is important to mention that the indices included in the groups of phylogenetic diversity and functional diversity, each explores a different facet of phylogenetic diversity (Kembel et al., 2010) and functional diversity (Laliberte et al., 2010), respectively.
</p>
<p><em>Rarity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species (species richness) in the sample, <em>s</em> is the number of samples, <code class="reqn">p_{ij}</code> is the presence or absence (0 or 1) of the species <em>i</em> in the sample <em>j</em>, <code class="reqn">r_{ij}</code> is the number of records or abundance of the species <em>i</em> in the sample <em>j</em>, <code class="reqn">R_{i}</code> is the total number of records or total abundance of the species <em>i</em> considering all samples, <code class="reqn">Q_i</code> is the occurrence of species <em>i</em>, <code class="reqn">Q_{min}</code> and <code class="reqn">Q_{max}</code> are respectively the minimum and maximum occurrences in the species pool, <em>r</em> is the chosen rarity cutoff point (as a percentage of occurrence), <code class="reqn">w_i</code> is the weight of the <em>ith</em> species in the assemblage, <code class="reqn">w_{min}</code> and <code class="reqn">w_{max}</code> the minimum and maximum weights respectively.
</p>
<p><em>Leroy</em> (Leroy et al., 2012; 2013)
</p>
<p style="text-align: center;"><code class="reqn">w_i=\frac{1}{e^{(\frac{Q_i-Q_{min}}{rQ_{max}Q_{min}}*0.97+1.05)^2}}</code>
</p>

<p style="text-align: center;"><code class="reqn">I_{RR}=\frac{\frac{\displaystyle\sum_{i=1}^{S}w_i}{S}-w_{min}}{w_{max}-w_{min}}</code>
</p>

<p><em>Geographical rarity</em> This index is a novel contribution of this package.
</p>
<p style="text-align: center;"><code class="reqn">GR=1-\frac{\displaystyle\sum_{i=1}^{S}\frac{\displaystyle\sum_{j=1}^{s}p_{ij}}{s}}{S}</code>
</p>

<p><em>Occurrential rarity</em> This index is a novel contribution of this package.
</p>
<p style="text-align: center;"><code class="reqn">OR=1-\frac{\displaystyle\sum_{i=1}^{S}\frac{\displaystyle\sum_{j=1}^{s}r_{ij}}{R_{i}}}{S}</code>
</p>

<p><em>Heterogeneity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species (species richness) in the sample, <code class="reqn">p_i</code> is the abundace's proportion of species <em>i</em>, <em>N</em> is the total number of individuals in the sample, <code class="reqn">n_i</code> is the number of individuals of the species <em>i</em>, <code class="reqn">n_{max}</code> is the number of individuals of the most abundant species, <em>a</em> and <em>q</em> are the orders of Renyi and Tsallis indices respectively and, finally, in Fisher's alpha the index is the <code class="reqn">\alpha</code> parameter.
</p>
<p><em>log Shannon-Wiener (S.W.LOG2)</em> and <em>ln Shannon-Wiener (S.W)</em> (Wiener, 1939; 1948; 1949; Shannon, 1948; Shannon &amp; Weaver, 1949). See Spellerger &amp; Fedor (2013) for an explanation of the dual use of the terms Shannon-Wiener and Shannon-Weaver to refer to this diversity index.
</p>
<p style="text-align: center;"><code class="reqn">H=-\displaystyle\sum_{i=1}^{S}p_ilog_2p_i</code>
</p>

<p style="text-align: center;"><code class="reqn">H'=-\displaystyle\sum_{i=1}^{S}p_ilnp_i</code>
</p>

<p><em>Fisher's alpha</em> (Fisher et al., 1943)
</p>
<p style="text-align: center;"><code class="reqn">{\alpha}x,\frac{{\alpha}x^2}{2},\frac{{\alpha}x3}{3},.....,\frac{{\alpha}x^n}{n}</code>
</p>

<p><em>Simpson</em> (Simpson, 1949) 
</p>
<p style="text-align: center;"><code class="reqn">D_1=1-\displaystyle\sum_{i=1}^{S}p_i^2</code>
</p>

<p><em>Inverse Simpson (InvSimpson)</em> Williams (1964)
</p>
<p style="text-align: center;"><code class="reqn">D_2=\frac{1}{\displaystyle\sum_{i=1}^{S}p_i^2}</code>
</p>

<p><em>Brillouin</em> (Brillouin, 1956)
</p>
<p style="text-align: center;"><code class="reqn">H_B=\frac{lnN!-\displaystyle\sum_{i=1}^{S}lnn_i!}{N}</code>
</p>

<p><em>Margalef</em> (Margalef, 1959)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mg}=\frac{(S-1)}{lnN}</code>
</p>

<p><em>Renyi entropy</em> (Renyi, 1961) 
</p>
<p style="text-align: center;"><code class="reqn">H_a=\frac{1}{(1-a)}log\displaystyle\sum_{i=1}^{S}p_i^a</code>
</p>
 
<p><em>Menhinick</em> (Menhinick, 1964)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mn}=\frac{S}{\sqrt{N}}</code>
</p>

<p><em>McIntosh</em> (McIntosh, 1967)
</p>
<p style="text-align: center;"><code class="reqn">D_{Mc}=\frac{N-\sqrt{\displaystyle\sum_{i=1}^{S}n_i^2}}{N-\sqrt{N}}</code>
</p>

<p><em>Inverse Berger-Parker (InvB.P)</em> (Berger &amp; Parker, 1970)
</p>
<p style="text-align: center;"><code class="reqn">D_{BP}=\frac{1}{\frac{n_{max}}{N}}</code>
</p>

<p><em>Hill numbers</em> (Hill, 1973)
</p>
<p>Hill-Renyi
</p>
<p style="text-align: center;"><code class="reqn">N_a=e^{H_a}</code>
</p>

<p>Hill-Tsallis
</p>
<p style="text-align: center;"><code class="reqn">N_q=(1-(q-1)H_q)^\frac{1}{(1-q)}</code>
</p>

<p>where
</p>
<p>a or q = 0 is species richness
</p>
<p>a or q = 1 is Shannon's index (H')
</p>
<p>a or q = 2 is Inverse Simpson's index (<code class="reqn">D_2</code>)
</p>
<p>a or q = Inf is Inverse Berger-Parker index (<code class="reqn">D_{BP}</code>)
</p>
<p><em>Tsallis entropy</em> (Patil &amp; Taillie, 1982; Tsallis, 1988)
</p>
<p style="text-align: center;"><code class="reqn">H_q=\frac{1}{(q-1)}(1-\displaystyle\sum_{i=1}^{S}p_i^q)</code>
</p>

<p><em>Evenness indices</em>
</p>
<p>Annotations of the equations as mentioned for heterogeneity indices.
</p>
<p><em>Simpson evenness (SimpsonE)</em> (Simpson, 1949) 
</p>
<p style="text-align: center;"><code class="reqn">E_{D1}=\frac{D_1}{S}</code>
</p>

<p><em>Pielou (PielouE)</em> (Pielou, 1966)
</p>
<p style="text-align: center;"><code class="reqn">J'=\frac{H'}{lnS}</code>
</p>

<p><em>McIntosh evenness (McIntoshE)</em> (McIntosh, 1967)
</p>
<p style="text-align: center;"><code class="reqn">E_{Mc}=\frac{N-\sqrt{\displaystyle\sum_{i=1}^{S}n_i^2}}{N-\frac{N}{\sqrt{S}}}</code>
</p>

<p><em>Hill evenness (HillE)</em> (Hill, 1973)
</p>
<p>It is used Hill-Renyi numbers where in <code class="reqn">N_2</code> the value of a = 2 and in <code class="reqn">N_2</code> the value of a = 1
</p>
<p style="text-align: center;"><code class="reqn">E_{2,1}=\frac{N_2}{N_1}</code>
</p>

<p><em>Heip evenness (HeipE)</em> (Heip, 1974)
</p>
<p style="text-align: center;"><code class="reqn">E_{Heip}=\frac{e^{H'}-1}{S-1}</code>
</p>

<p><em>Camargo (CamargoE)</em> (Camargo, 1992)
</p>
<p style="text-align: center;"><code class="reqn">E_{Camargo}=1-\displaystyle\sum_{i=1}^{S}\displaystyle\sum_{j=i'+1}^{S}\frac{|p_i-p_j|}{s}</code>
</p>

<p><em>Smith and Wilson's Index (Evar)</em> (Smith and Wilson, 1996)
</p>
<p style="text-align: center;"><code class="reqn">E_{var}=1-\left(\frac{2}{\pi}\right)\left(arctan\left(\frac{\displaystyle\sum_{i=1}^{S}{\left(lnn_i-\frac{\displaystyle\sum_{j=1}^{S}n_j}{S}\right)}^2}{S}\right)\right)</code>
</p>

<p><em>Taxonomic diversity indices</em>
</p>
<p>In the following equations summation goes over species <em>i</em> and <em>j</em>, <code class="reqn">\omega</code> are the taxonomic distances among taxa, <em>x</em> are species abundances, and <em>n</em> is the total abundance for a site.
</p>
<p><em>Taxonomic diversity (D)</em> (Clarke, 1995; 1998; 2001):
</p>
<p style="text-align: center;"><code class="reqn">\Delta=\frac{\sum\sum_{i&lt;j}\omega_{ij}x_ix_j}{n(n-1)/2}</code>
</p>

<p><em>Taxonomic distinctness (Dstar)</em> (Clarke and Warwick, 1998):
</p>
<p style="text-align: center;"><code class="reqn">\Delta^*=\frac{\sum\sum_{i&lt;j}\omega_{ij}x_ix_j}{\sum\sum_{i&lt;j}x_ix_j}</code>
</p>

<p><em>Phylogenetic diversity indices</em>
</p>

<table>
<tr>
<td style="text-align: left;">
Faith's phylogenetic diversity </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Faith (1992)</td>
</tr>
<tr>
<td style="text-align: left;">
Mean pairwise phylogenetic distance </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Webb et al. (2008)</td>
</tr>
<tr>
<td style="text-align: left;">
Mean nearest taxon distance </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Webb et al. (2008)</td>
</tr>
<tr>
<td style="text-align: left;">
Phylogenetic species richness </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
<td style="text-align: left;">
Phylogenetic species variability </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
<td style="text-align: left;">
Phylogenetic species evenness </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
<td style="text-align: left;">
Phylogenetic species clustering </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Helmus et al. (2007)</td>
</tr>
<tr>
<td style="text-align: left;">
Quadratic entropy </td>
<td style="text-align: left;">  </td>
<td style="text-align: left;">   </td>
<td style="text-align: left;"> Rao (1982)
</td>
</tr>
</table>
<p><em>Functional diversity indices</em>
</p>
<p>In the following equations <em>S</em> is the number of species, where <code class="reqn">d_{ij}</code> is the difference between the <em>i-th</em> and <em>j-th</em> species, <code class="reqn">p_i</code> is the abundance's proportion of the species <em>i</em>, <code class="reqn">p_j</code> is the abundance's proportion of the species <code class="reqn">j</code>, <em>EW</em> is weighted evenness, dist(<em>i</em>, <em>j</em>) is the Euclidean distance between species <em>i</em> and <em>j</em>, the species involved is branch <em>l</em>, <code class="reqn">w_i</code> is the relative abundance of the species <em>i</em>, <code class="reqn">PEW_l</code> is the partial weighted evenness, <code class="reqn">g_k</code> are the coordinates of the center of gravity of the <em>V</em> species forming the vertices of the convex hull, <code class="reqn">x_{ik}</code> is the coordinate of species <em>i</em> on trait <em>k</em>, <code class="reqn">dG_i</code> is the Euclidean distance of the center of gravity, <code class="reqn">\bar{dG}</code> is the mean distance of the <em>S</em> species to the center of gravity, <code class="reqn">\Delta</code><em>d</em> is the sum of abundance-weighted deviances and <code class="reqn">\Delta</code><em>|d|</em> absolute abundance-weighted deviances.
</p>
<p><em>Rao's quadratic entropy</em> (Rao, 1982; Botta-Dukat, 2005):
</p>
<p style="text-align: center;"><code class="reqn">Q=\displaystyle\sum_{i=1}^{S-1}\displaystyle\sum_{j=i+1}^{S-1}d_{ij}p_ip_j</code>
</p>

<p style="text-align: center;"><code class="reqn">d_{ij}=\displaystyle\sum_{k=1}^{n}\displaystyle\sum_{l=1}^{n}w_{kl}(X_{ik}-X_{jk})(X_{il}-X_{jl})</code>
</p>

<p><em>Functional group richness (FGR)</em> Petchey and Gaston (2006)
</p>
<p><em>Functional richness (FRic)</em> is measured as the amount of functional space (convex hull volume) filled by the community (Villeger et al., 2008).
</p>
<p><em>Functional evenness (FEve)</em> (Villeger et al., 2008):
</p>
<p style="text-align: center;"><code class="reqn">FEve=\frac{\displaystyle\sum_{l=1}^{S-1}min(PEW_l,\frac{1}{S-1})-\frac{1}{S-1}}{1-\frac{1}{S-1}}</code>
</p>


<table>
<tr>
<td style="text-align: left;">
<code class="reqn">PEW_l=\frac{EW_l}{\displaystyle\sum_{l=1}^{S}EW_l}</code> </td>
<td style="text-align: left;"> <code class="reqn">EW_l=\frac{dis(i,j)}{w_i+w_j}</code>
</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p><em>Functional divergence (FDiv)</em> (Villeger et al., 2008):
</p>
<p style="text-align: center;"><code class="reqn">FDiv=\frac{\Delta{d}+\bar{dG}}{\Delta{|d|}+\bar{dG}}</code>
</p>


<table>
<tr>
<td style="text-align: left;">
<code class="reqn">\Delta{|d|}=\displaystyle\sum_{i=1}^{S}w_i|dG_i-\bar{dG}|</code> </td>
<td style="text-align: left;"> <code class="reqn">\Delta{d}=\displaystyle\sum_{i=1}^{S}w_i(dG_i-\bar{dG})</code>
</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<table>
<tr>
<td style="text-align: left;">
<code class="reqn">\bar{dG}=\frac{1}{S}\displaystyle\sum_{i=1}^{S}dG_i</code> </td>
<td style="text-align: left;"> <code class="reqn">dG_i=\sqrt{\displaystyle\sum_{k=1}^{T}(x_{ik}-g_k)^2}</code> </td>
<td style="text-align: left;"> <code class="reqn">g_k=\frac{1}{V}\displaystyle\sum_{i=1}^{V}x_{ik}</code>
</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p><em>Functional dispersion (FDis)</em> (Laliberte and Legendre, 2010):
</p>

<table>
<tr>
<td style="text-align: left;">
<code class="reqn">FDisp=\frac{\sum{a_jz_j}}{\sum{a_j}}</code> </td>
<td style="text-align: left;"> <code class="reqn">c=\frac{\sum{a_jx_{ij}}}{\sum{a_j}}</code>
</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p><strong>2.</strong> Each index is transformed to a scale range between 0 and 1 for all samples with the following equation:
</p>
<p style="text-align: center;"><code class="reqn">\frac{(index of the sample-min)}{(max-min)}</code>
</p>

<p>where <em>min</em> and <em>max</em> are the minimum and maximum values of the index considering all samples, respectively.
</p>
<p><strong>3.</strong> With the standardized values of the indices, the algorithm calculates the polar coordinates of all samples with all possible combinations among all groups of indices. Therefore, in each combination an index of each group of rarity, heterogeneity (species richness is included in this group), evenness, taxonomic/ phylogenetic diversity and functional diversity (if it is included functional traits in the analysis) is used for calculating the polar coordinates of all samples. In the group of taxonomic/phylogenetic diversity the user must use either taxonomy or a phylogenetic tree, so either taxonomic diversity or phylogenetic diversity indices are used in the algorithm. The X and Y polar coordinates for each sample are estimated using the following equations:
</p>

<table><tr>
<td style="text-align: left;">
<code class="reqn">X=\displaystyle\sum_{i=1}^{4}|z_j|cos(\alpha)</code> </td>
<td style="text-align: left;"> <code class="reqn">Y=\displaystyle\sum_{i=1}^{4}|z_j|sin(\alpha)</code>
</td>
</tr></table>
<p>where <em>z</em> is the standardized value of the index <em>j</em> of the four groups considered.
</p>
<p>Each index is assigned an angle (<code class="reqn">\alpha</code>). Degrees to radians angle conversion is carried out assuming that 1 degree = 0.0174532925 radians.
</p>
<p><strong>4.</strong> With the polar coordinates of the samples obtained for each combination, it is calculated the convex hull (alpha = 6) and the mean Euclidean distance, and the values are saved in a file.
</p>
<p><strong>5.</strong> The algorithm selects the combination of indices with the highest value of the mean between convex hull and mean Euclidean distance among samples, therefore priority is given to maximize dispersion among samples (see Fig. 1). The polar coordinates of the selected combination are depicted on a diagram, where it is possible to see the differences in rarity, heterogeneity, evenness and taxonomic/phylogenetic  diversity and/or functional diversity (if it is included) among assemblages.
</p>
<p><strong>6.</strong> Finally, DER function allows the user to select the four/five indices to be used in the diagram, so the algorithm of selecting the combination with the maximum dispersion among samples is not applied.
</p>
<p><strong>FUNCTIONS</strong>
</p>
<p>The index Fisher alpha was estimated with the function fisher.alpha, the index Renyi with the function renyi, the index Tsallis with the function tsallis, the taxonomic diversity and taxonomic distinctness with the functions taxa2dist and taxondive, all of them of the package vegan (Oksanen et al., 2016). The ellipse is depicted with the function plotellipse of the package shape (Soetaert, 2016). The convex hull (alpha=6) was calculated with the function areapl of the package splancs (Bivand et al., 2016). The color legend of DER plot was depicted with the function color.legend of the package plotrix (Lemon et al., 2016). The rarity index of Leroy was calculated with the functions rWeights and Irr, both of the package Rarity (Leroy et al., 2012; 2103; Leroy, 2016). The functional diversity indices were calculated with the function dbFD of the package FD (Laliberte et al., 2015). The phylogenetic indices were calculated with the functions psv, psr, pse, psc, raoD, mntd, mpd and pd of the package picante (Kembel et al. 2010 2016)
</p>
<p><strong>EXAMPLE</strong>
</p>
<p>The example without functional diversity is a dataset with the abundance of rotifers species in ponds (see table 1 in Mazuelos et al., 1993). In the argument <em>Index</em> were selected Rarity, Menhinick, McIntoshE and Dstar, which are the indices selected by the algorithm when <em>Index=NULL</em> (default option). The sample G3.1 had the lowest values of the indices of rarity, heterogeneity, evenness and taxonomic diversity and the pond I3.1 the highest values for all indices.
</p>

<table><tr>
<td style="text-align: center;">
<img src="../help/figures/EcoIndRF2.jpg" width="50" alt="EcoIndRF2.jpg">
</td>
</tr></table>
<h3>Value</h3>

<p>It is depicted a plot of polar coordinates estimated with the rarity, heterogeneity, evenness, taxonomic/phylogenetic and functional diversity indices, CSV files are saved with all the indices, the polar coordinates estimated with the indices specified in the argument <em>Index</em> or estimated by the algorithm, and the area of the convex hull and mean Euclidean distance obtained in the polar coordinates system for all combinations of the indices.</p>


<h3>References</h3>

<p>Berger, W.H., Parker, F.L. (1970) Diversity of planktonic Foramenifera in deep sea sediments. <em>Science</em>, 168: 1345-1347.
</p>
<p>Bivand, R., Rowlingson, B., Diggle, P., Petris, G., Eglen, S. (2016) Spatial and Space-Time Point Pattern Analysis. R Package Version 2.01-39. <a href="https://CRAN.R-project.org/package=splancs">https://CRAN.R-project.org/package=splancs</a>
</p>
<p>Botta-Dukat, Z. (2005) Rao's quadratic entropy as a measure of functional diversity based on multiple traits. <em>Journal of Vegetation Science</em>, 16: 533-540.
</p>
<p>Brillouin, L. (1956) <em>Science and information theory</em>. New York: Academic Press.
</p>
<p>Camargo, J.A. (1992) New diversity index for assessing structural alterations in aquatic communities. <em>Bulletin of Environmental Contamination and Toxicology</em>, 48, 428-434.
</p>
<p>Clarke, K.R. &amp; Warwick, R.M. (1998). A taxonomic distinctness index and its statistical properties. <em>Journal of Applied Ecology</em>, 35: 523-531.
</p>
<p>Faith, D.P. (1992) Conservation evaluation and phylogenetic diversity. <em>Biological Conservation</em>, 61: 1-10.
</p>
<p>Fisher, R.A., Corbet, A.S. &amp; Williams, C.B. (1943) The relation between the number of species and the number of individuals in a random sample of animal population. <em>Journal of Animal Ecology</em>, 12: 42-58.
</p>
<p>Heip, C. 1974. A new index measuring evenness. <em>Journal of the Marine Biological Association of the United Kingdom</em>, 54: 555-557.
</p>
<p>Helmus, M.R., Bland, T.J., Williams, C.J., Ives, A.R. (2007) Phylogenetic measures of biodiversity. <em>The American Naturalist</em>, 169: E68-E83.
</p>
<p>Hill, M.O. (1973) Diversity and evenness: a unifying notation and its consequences. <em>Ecology</em>, 54: 427-432.
</p>
<p>Hurlbert, S.H. 1971. The nonconcept of species diversity: a critique and alternative parameters. <em>Ecology</em>, 52: 577-586.
</p>
<p>Kembel, S.W., Cowan, P.D., Helmus, M.R., Cornwell, W.K., Morlon, H., Ackerly, D.D., Blomberg, S.P.,Webb, C.O. (2010) Picante: R tools for integrating phylogenies and ecology. <em>Bioinformatics</em>, 26: 1463-1464.
</p>
<p>Kembel, S.W., Ackerly, D.D., Blomberg, S.P., Cornwell, W.K., Cowan, P.D., Helmus, M.R., Morlon, H. &amp; Webb, C.O. (2016) R tools for integrating phylogenies and ecology. R Package Version 1.6-2. <a href="https://CRAN.R-project.org/package=picante">https://CRAN.R-project.org/package=picante</a>
</p>
<p>Laliberte, E. &amp; Legendre, P. (2010) A distance-based framework for measuring functional diversity from multiple traits. <em>Ecology</em>, 91: 299-305.
</p>
<p>Laliberte, E., Legendre, P. &amp; Shipley, B. (2015) Measuring functional diversity (FD) from multiple traits, and other tools for functional ecology. R package version 1.0-12. Available at: <a href="https://CRAN.R-project.org/package=FD">https://CRAN.R-project.org/package=FD</a>.
</p>
<p>Lemon, J., Bolker, B., Oom, S., Klein, E., Rowlingson, B.,Wickham, H., Tyagi, A., Eterradossi, O., Grothendieck, G., Toews, M., Kane, J., Turner, R., Witthoft, C., Stander, J., Petzoldt, T., Duursma, R., Biancotto, E., Levy, O., Dutang, C., Solymos, P., Engelmann, R., Hecker, M., Steinbeck, F., Borchers, H., Singmann, H., Toal, T. &amp; Ogle, D. (2016) Various plotting functions. R package version 3.6-3. Available at: <a href="https://CRAN.R-project.org/package=plotrix">https://CRAN.R-project.org/package=plotrix</a>.
</p>
<p>Leroy, B., Petillon, J., Gallon, R., Canard, A. &amp; Ysnel, F. (2012) Improving occurrence-based rarity metrics in conservation studies by including multiple rarity cut-off points. <em>Insect Conservation and Diversity</em>, 5, 159-168.
</p>
<p>Leroy, B., Canard, A. &amp; Ysnel, F. (2013) Integrating multiple scales in rarity assessments of invertebrate taxa. <em>Diversity and Distributions</em>, 19, 794-803.
</p>
<p>Leroy, B. (2016) Calculation of Rarity Indices for Species and Assemblages of Species. R package version 1.3-4. Available at: <a href="https://CRAN.R-project.org/package=Rarity">https://CRAN.R-project.org/package=Rarity</a>.
</p>
<p>Oksanen, J., Blanchet, F.G., Kindt, R., Legendre, P., Minchin, P.R., O'Hara, R.B., Simpson, G.L., Solymos, P., Henry, M., Stevens, H. &amp; Wagner, H. (2016) Community Ecology Package. R package version 2.4-0. Available at: <a href="https://CRAN.R-project.org/package=vegan">https://CRAN.R-project.org/package=vegan</a>.
</p>
<p>Patil, G.P. &amp; Taillie, C. (1982) Diversity as a concept and its measurement. <em>Journal of the Acoustical Society of America</em>, 77: 548-561.
</p>
<p>Pielou, E.C. (1966) The measurement of diversity in different types of biological collections. <em>Journal Theoretical Biology</em>, 13: 131-144.
</p>
<p>Petchey, O.L. &amp; Gaston, K.J. (2002) Functional diversity (FD), species richness and community composition. <em>Ecology Letters</em>, 5: 402-411.
</p>
<p>Margalef, R. (1958) Information theory in ecology. <em>General Systems</em>, 3: 36-71.
</p>
<p>Mazuelos, N., Toja, J. &amp; Guisande, C. (1993) Rotifers in ephemeral ponds of Doñana National Park. <em>Hydrobiologia</em>, 255/256: 429-434.
</p>
<p>McIntosh, R.P. (1967) An index of diversity and the relation of certain concepts to diversity. <em>Ecology</em>, 48: 392-404.
</p>
<p>Menhinick, E.P. (1964) A comparison of some species-individual diversity indices applies to samples of field insects. <em>Ecology</em>, 45: 859-861.
</p>
<p>Renyi, A. (1961) On measures of information and entropy. <em>Proceedings of the fourth Berkeley Symposium on Mathematics, Statistics and Probability Vol. 1</em>: 547-561.
</p>
<p>Rao, C.R. (1982) Diversity and dissimilarity coefficients: a unified approach. <em>Theoretical Population Biology</em>, 21: 24-43.
</p>
<p>Shannon, C.E. (1948) A Mathematical Theory of Communication. <em>Bell System Technical Journal</em>, 27 (4): 379-423, 623-656.
</p>
<p>Shannon, C.E. &amp; Weaver, W. (1949) <em>The Mathematical Theory of Communication</em>. The University of Illinois Press, Illinois. 
</p>
<p>Simpson, E.H. (1949) Measurement of diversity. <em>Nature</em>, 163: 688.
</p>
<p>Smith, B. &amp; J.B. Wilson (1996) A consumer's guide to evenness indices. <em>Oikos</em>, 76: 70-82.
</p>
<p>Soetaert, K. 2016. Functions for plotting graphical shapes, colors. R package version 1.4.2. Available at: <a href="https://CRAN.R-project.org/package=shape">https://CRAN.R-project.org/package=shape</a>.
</p>
<p>Spellerger, I.F. &amp; Fedor P.J. (2013) A tribute to Claude Shannon (1916-2001) and a plea for more rigorous use of species richness, species diversity and the "Shannon-Wiener" Index. <em>Global Ecology and Biogeography</em>, 12: 177-179.
</p>
<p>Tsallis, C. (1988) Possible generalization of Boltzmann-Gibbs statistics. <em>Journal of Statistical Physics</em>, 52: 479-487.
</p>
<p>Villeger, S., Mason, N.W.H. &amp; Mouillot, D. (2008) New multidimensional functional diversity indices for a multifaceted framework in functional ecology. <em>Ecology</em> 89: 2290-2301.
</p>
<p>Webb, C.O., Ackerly, D.D. &amp; Kembel, W. (2008) Phylocom: software for the analysis of phylogenetic community structure and trait evolution. <em>Bioinformatics</em>, 24: 2098-2100.
</p>
<p>Wiener, N. (1939) The ergodic theorem. <em>Duke Mathematical Journal</em>, 5: 1-18.
</p>
<p>Wiener, N. (1948) <em>Cybernetics</em>. Wiley, New York.
</p>
<p>Wiener, N. (1949) <em>The interpolation, extrapolation, and smoothing of stationary time series</em>. Wiley, New York.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#An example without functional diversity

data(Rotifers)

DER(data=Rotifers, Samples=c("J1.1","K4.1","G3.1","F2.1","K2.2","F8.2","F8.1",
"F1.1","F4.1","J2.1","E5.1","H5.1","K3.2","E4.2","I6.1","K2.1","J5.1","I3.1",
"K3.3","G5.1","E6.1","J1.2","J6.1","G7.1","G6.1","G4.1","E3.1","E4.3","E2.1",
"H6.2","F7.1","J6.2"), Species="Species", Taxon=c("Class","Subclass",
"Superorder","Order","Family","Genus"), pos="Pos", Index=c("Rarity.G","Menhinick",
"McIntoshE", "Dstar"), save=FALSE)



</code></pre>


</div>