<div class="container">

<table style="width: 100%;"><tr>
<td>hmm.search</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Searching for motifs using HMM searches</h2>

<h3>Description</h3>

<p>This function uses MAFFT and HMMER to search for sequences with RxLR or CRN motifs using hidden markov models.
</p>


<h3>Usage</h3>

<pre><code class="language-R">hmm.search(original.seq, regex.seq, alignment.file = NULL,
  save.alignment = FALSE, mafft.path = NULL, num.threads = 2,
  hmm.path = NULL, seed = 12345)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>original.seq</code></td>
<td>
<p>The absolute path for the original six-frame translation FASTA file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>regex.seq</code></td>
<td>
<p>A list of <code>SeqFastadna</code> objects resulting from <code>regex.search</code>. The HMM profile will be constructed using these sequences</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alignment.file</code></td>
<td>
<p>(Optional) The absolute path for an alignment file of the sequences to build the hmmer profile from. It's recommended that the alignment file cointains the same sequences than the ‘regex.seq' files. If the user provides the absolute path, *effectR* won’t use MAFFT to align the sequences and will use the alignment for the HMMER searches. If no alignment file is provided, *effectR* will use MAFFT to align the sequences from 'regex.seq' and run HMMER.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.alignment</code></td>
<td>
<p>(Optional) Save the alignment in the returning object. The MAFFT alignment will be saved as the first element of the returned object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mafft.path</code></td>
<td>
<p>(Optional) Local path of folder containing the MAFFT binary executable file or the executable file itself. If not specified, then MAFFT must be in the program search path.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.threads</code></td>
<td>
<p>(Optional) Number of threads to be used by MAFFT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hmm.path</code></td>
<td>
<p>(Optional) Local path of  folder containing the HMMER binaries.  If not specified, then HMMER executables must be in the program search path.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>(Optional) The seed to used with HMMER commands. Set this to get the same output each time</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>hmm.search</code> uses the results from <code>regex.search</code> to search for motifs of interest using Hidden Markov Models after aligning the sequences with MAFFT.
After the multiple sequence alignment is complete, the function constructs a HMM profile using the alignment data. The HMM profile is in the original list of <code>SeqFastadna</code> objects to obtain the best HMM results with sequences with RxLR or CRN motifs.
</p>


<h3>Value</h3>

<p>A list of three elements: REGEX candidate effectors, HMM candidate effectors, and HMM results table.
</p>


<h3>Note</h3>

<p>If MAFFT/HMMER are not the program search path, the user has to specify the path for the MAFFT and the HMMER executable binaries and specify them in the <code>mafft.path</code> and <code>hmm.path</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 

fasta.file &lt;- system.file("extdata", "test_infestans.fasta", package = "effectR")
ORF &lt;- seqinr::read.fasta(fasta.file)
REGEX &lt;- regex.search(ORF, motif="RxLR")
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq = REGEX,
 alignment.file=NULL, save.alignment=T)

# To save the alignment:
library(seqinr)
write.fasta(sequences = getSequence(candidate.rxlr$Alignment),
 names = getName(candidate.rxlr$Alignment),
 file.out = "Alin_infestans.fasta", nbchar = 10000)

# To use an alignment file
ALIN &lt;- system.file("extdata", "Alin_infestans.fasta", package = "effectR")
candidate.rxlr &lt;- hmm.search(original.seq = fasta.file, regex.seq = REGEX,
 alignment.file = ALIN)

## End(Not run)
</code></pre>


</div>