<div class="container">

<table style="width: 100%;"><tr>
<td>eGST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run eGST.</h2>

<h3>Description</h3>

<p>Run eGST to estimate the posterior probability that the genetic
susceptibility of the phenotype of
an individual in the study is mediated through eQTLs specific to a tissue of interest.
To create sets of tissue-specific eQTLs in your context, please see our manuscript:
Majumdar A, Giambartolomei C, Cai N, Freund MK, Haldar T, J Flint, Pasaniuc B (2019)
Leveraging eQTLs to identify tissue-specific genetic subtype of complex trait, bioRxiv.
</p>


<h3>Usage</h3>

<pre><code class="language-R">eGST(pheno, geno, tissues, logLimprovement = 5 * 10^(-8),
  seed_choice = sample(1:1000, size = 1), nIter = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pheno</code></td>
<td>
<p>A numeric vector of length N where N is the number of individuals. It
contains the GWAS phenotype values of individuals. No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>A list with K elements where K is the number of tissues. Each element of
geno is the genotype matrix of the eQTLs specific to a tissue in the GWAS cohort.
So j-th element of geno is N by Mj matrix
containing the genotype data of N individuals (rows) at Mj eQTLs (columns) specific to j-th tissue.
Each eQTL is a bi-allelic SNP with minor allele frequency &gt; 0.01.
Genotypes at each eQTL must be normalized across N individuals.
If 0/1/2 valued genotype matrix is provided, it is internally normalized. No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tissues</code></td>
<td>
<p>A character vector of length K. It contains the names of tissues of interest in the analysis.
The order of tissues in this vector must match the order of tissues in the previous argument 'geno'. No default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logLimprovement</code></td>
<td>
<p>A positive real number specifying the minimum possible improvement
of data log-likelihood in MAP-EM stopping criterion. Default <code class="reqn">5*10^(-8)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed_choice</code></td>
<td>
<p>An integer providing the choice of random seed for initialization in MAP-EM algorithm.
Default is an integer randomly selected in (1,...,1000).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nIter</code></td>
<td>
<p>An integer providing the maximum number of iterations allowed in the MAP-EM algorithm. Default is 100.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output produced by <code>eGST</code> is a list which consists of various components.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>gamma</code></td>
<td>
<p>A N by K matrix providing the tissue-specific posterior probability of N individuals across K tissues.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alfa</code></td>
<td>
<p>Baseline tissue-specific intercepts/means of the trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>Tissue-specific eQTLs' genetic effect on the trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma_g</code></td>
<td>
<p>Square root of the variance of tissue-specific per-eQTL genetic effect on the trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma_e</code></td>
<td>
<p>Square root of the error variance of tissue-specific subtype of the trait
which remains unexplained by the tissue-specific eQTLs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>Number of tissue-specific eQTLs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logL</code></td>
<td>
<p>log-likelihood of the data.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>A Majumdar, C Giambartolomei, N Cai, MK Freund, T Haldar, T Schwarz, J Flint, B Pasaniuc (2019) Leveraging eQTLs to identify
tissue-specific genetic subtype of complex trait. bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ExamplePhenoData)
pheno &lt;- ExamplePhenoData
head(pheno)
data(ExampleEQTLgenoData)
geno &lt;- ExampleEQTLgenoData
geno[[1]][1:5,1:5]
geno[[2]][1:5,1:5]
tissues &lt;- paste("tissue", 1:2, sep = "")
result &lt;- eGST(pheno, geno, tissues)
str(result)

</code></pre>


</div>