<div class="container">

<table style="width: 100%;"><tr>
<td>encode_genotype_vec</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Encode a genotype/SNP vector</h2>

<h3>Description</h3>

<p>Standardises homozygous SNP alleles (e.g. recorded as 'A') to two character
form (e.g. 'A/A') and orders heterozygous SNP alleles alphabetically (e.g.
"GA" becomes "A/G"). The SNP values are then converted from a character
vector to an ordered factor, ordered by SNP allele frequency (e.g. most
frequent SNP allele is 1, second most frequent value is 2, and least frequent
values is 3). This method embeds the numeric relationship between the SNP
allele frequencies while preserving value labels.
</p>


<h3>Usage</h3>

<pre><code class="language-R">encode_genotype_vec(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>input vector containing genotype data</p>
</td>
</tr></table>
<h3>Value</h3>

<p>Ordered factor, ordered by allele frequency in variable
</p>


</div>