<div class="container">

<table style="width: 100%;"><tr>
<td>genSimDat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Generate Simulated Data
</h2>

<h3>Description</h3>

<p>Generate a simulated dataset from a mixture of Bayesian hierarchical models
with two covariates: age and sex.
</p>


<h3>Usage</h3>

<pre><code class="language-R">genSimDat(G, n, psi, t_pi, m.age = 50, sd.age = 5, p.female = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>integer. Number of probes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>integer. Number of samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi</code></td>
<td>

<p>numeric. A vector of model hyper-parameters with elements
<code class="reqn">\alpha_1</code>, <code class="reqn">\beta_1</code>, <code class="reqn">k_1</code>, 
<code class="reqn">\eta_{1,intercept}</code>, <code class="reqn">\eta_{1,age}</code>, <code class="reqn">\eta_{1,sex}</code>,
<code class="reqn">\alpha_2</code>, <code class="reqn">\beta_2</code>, <code class="reqn">k_2</code>, 
<code class="reqn">\eta_{2,intercept}</code>, <code class="reqn">\eta_{2,age}</code>, <code class="reqn">\eta_{2,sex}</code>,
<code class="reqn">\alpha_3</code>, <code class="reqn">\beta_3</code>, <code class="reqn">k_3</code>, 
<code class="reqn">\eta_{3,intercept}</code>, <code class="reqn">\eta_{3,age}</code>, <code class="reqn">\eta_{3,sex}</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t_pi</code></td>
<td>

<p>numeric. A vector of mixture proportions: <code class="reqn">\pi_1</code> (proportion
for probes over-expressed in cases); 
<code class="reqn">\pi_2</code> (proportion for probes under-expressed in cases).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m.age</code></td>
<td>

<p>numeric. mean age.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd.age</code></td>
<td>

<p>numeric. standard deviation of age.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.female</code></td>
<td>

<p>numeric. proportion of females.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An ExpressionSet object.
</p>


<h3>Note</h3>

<p>Age will be mean-centered and scaled so that it will
have mean zero and variance one.
</p>


<h3>Author(s)</h3>

<p>Yixin Zhang <a href="mailto:zhyl133@gmail.com">zhyl133@gmail.com</a>,
Wei Liu <a href="mailto:liuwei@mathstat.yorku.ca">liuwei@mathstat.yorku.ca</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@sanofi.com">weiliang.qiu@sanofi.com</a>
</p>


<h3>References</h3>

<p>Zhang Y, Liu W, Qiu W.
A model-based clustering via mixture of hierarchical models with covariate adjustment for detecting differentially expressed genes from paired design.
<em>BMC Bioinformatics</em> 24, 423 (2023)
</p>


<h3>Examples</h3>

<pre><code class="language-R">
set.seed(1234567)

true.psi = c(2, 1, 0.8,
             0.1, -0.01, -0.1,
             2, 1, 0.8,
             -0.1, -0.01, -0.1,
             2, 1, 0.8,
             -0.01, -0.1)
names(true.psi)=c("alpha1", "beta1", "k1",
                  "eta1.intercept", "eta1.Age", "eta1.Sex",
                  "alpha2", "beta2", "k2",
                  "eta2.intercept", "eta2.Age", "eta2.Sex",
                  "alpha3", "beta3", "k3",
                  "eta3.Age", "eta3.Sex")
true.pi=c(0.1, 0.1)
names(true.pi)=c("pi.OE", "pi.UE")
par.true=c(true.pi, true.psi)

esDiff = genSimDat(G = 1000, 
	       n = 20, 
	       psi = true.psi, 
	       t_pi = true.pi, 
	       m.age = 0, # scaled age 
	       sd.age = 1, # scaled age
	       p.female = 0.5)

print(esDiff)

</code></pre>


</div>