<div class="container">

<table style="width: 100%;"><tr>
<td>emBayes-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Robust Bayesian Variable Selection via Expectation-Maximization</h2>

<h3>Description</h3>

<p>This package provides the implementation of the spike-and-slab quantile LASSO (ssQLASSO) and spike-and-slab quantile group LASSO varying coefficient mixed model (ssQVCM) which combines the strength of Bayesian robust variable selection and the Expectation-Maximization (EM) coordinate descent approach. The alternative methods spike-and-slab LASSO (ssLASSO) and spike-and-slab group LASSO varying coefficient mixed model (ssVCM) are also included in the package.
</p>


<h3>Details</h3>

<p>Two user friendly, integrated interface <strong>cv.emBayes()</strong> and <strong>emBayes()</strong> allows users to flexibly choose the variable selection method by specifying the following parameter:
</p>

<table>
<tr>
<td style="text-align: right;">
quant: </td>
<td style="text-align: left;"> to specify different quantiles when using robust methods.</td>
</tr>
<tr>
<td style="text-align: right;"></td>
</tr>
<tr>
<td style="text-align: right;">
func: </td>
<td style="text-align: left;"> the model to perform variable selection. Four choices are available: </td>
</tr>
<tr>
<td style="text-align: right;"> </td>
<td style="text-align: left;"> "ssLASSO", "ssQLASSO", "ssVCM" and "ssQVCM". </td>
</tr>
<tr>
<td style="text-align: right;"></td>
</tr>
<tr>
<td style="text-align: right;">
error: </td>
<td style="text-align: left;"> to specify the difference between expectations of likelihood of two  </td>
</tr>
<tr>
<td style="text-align: right;"> </td>
<td style="text-align: left;"> consecutive iterations. It can be used to determine convergence.</td>
</tr>
<tr>
<td style="text-align: right;"></td>
</tr>
<tr>
<td style="text-align: right;">
maxiter: </td>
<td style="text-align: left;"> to specify the maximum number of iterations.
</td>
</tr>
</table>
<p>Function cv.emBayes() returns cross-validation errors based on the check loss, least squares loss and Schwarz Information Criterion along with the corresponding optimal tuning parameters.
Function emBayes() returns the estimated intercept, clinical coefficients, beta coefficients, scale parameter, probability parameter, number of iterations and expectation of likelihood at each iteration.
</p>


<h3>References</h3>

<p>Liu, Y., Ren, J., Ma, S., and Wu, C. (2024). The Spike-and-Slab Quantile LASSO for Robust Variable Selection in Cancer Genomics Studies.
<em>Statistics in Medicine.</em>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y., and Wu, C. (2022). Robust Bayesian variable selection for gene–environment interactions.
<em>Biometrics.</em> <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang,Y. and Wu, C. (2019). Robust network-based regularization
and variable selection for high dimensional genomics data in cancer prognosis.
<em>Genet. Epidemiol.</em>, 43:276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Wu, C., Zhang, Q., Jiang,Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi)genetic measurements.
<em>J Multivar Anal.</em>, 168:119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Tang, Z., Shen, Y., Zhang, X., and Yi, N. (2017). The spike-and-slab lasso generalized linear models for prediction and associated genes detection. 
<em>Genetics</em>, 205(1), 77-88 <a href="https://doi.org/10.1534/genetics.116.192195">doi:10.1534/genetics.116.192195</a>
</p>
<p>Tang, Z., Shen, Y., Zhang, X., and Yi, N. (2017). The spike-and-slab lasso Cox model for survival prediction and associated genes detection. 
<em>Bioinformatics</em>, 33(18), 2799-2807 <a href="https://doi.org/10.1093/bioinformatics/btx300">doi:10.1093/bioinformatics/btx300</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a> 
</p>
<p>Zhou, Y. H., Ni, Z. X., and Li, Y. (2014). Quantile regression via the EM algorithm. 
<em>Communications in Statistics-Simulation and Computation</em>, 43(10), 2162-2172 <a href="https://doi.org/10.1080/03610918.2012.746980">doi:10.1080/03610918.2012.746980</a>
</p>
<p>Ročková, V., and George, E. I. (2014). EMVS: The EM approach to Bayesian variable selection. 
<em>Journal of the American Statistical Association</em>, 109(506), 828-846 <a href="https://doi.org/10.1080/01621459.2013.869223">doi:10.1080/01621459.2013.869223</a>
</p>
<p>Li, Q., Lin, N., and Xi, R. (2010). Bayesian regularized quantile regression.
<em>Bayesian Analysis</em>, 5(3), 533-556 <a href="https://doi.org/10.1214/10-BA521">doi:10.1214/10-BA521</a>
</p>
<p>George, E. I., and McCulloch, R. E. (1993). Variable selection via Gibbs sampling. 
<em>Journal of the American Statistical Association</em>, 88(423), 881-889 <a href="https://doi.org/10.1080/01621459.1993.10476353">doi:10.1080/01621459.1993.10476353</a>
</p>


<h3>See Also</h3>

<p><code>cv.emBayes</code> <code>emBayes</code>
</p>


</div>