<div class="container">

<table style="width: 100%;"><tr>
<td>select_genes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Selects genes using the EMMIXgene algorithm.</h2>

<h3>Description</h3>

<p>Follows the gene selection methodology of 
G. J. McLachlan, R. W. Bean, D. Peel; A mixture model-based approach to the 
clustering of microarray expression data , Bioinformatics, Volume 18, 
Issue 3, 1 March 2002, Pages 413â€“422,
https://doi.org/10.1093/bioinformatics/18.3.413
</p>


<h3>Usage</h3>

<pre><code class="language-R">select_genes(
  dat,
  filename,
  random_starts = 4,
  max_it = 100,
  ll_thresh = 8,
  min_clust_size = 8,
  tol = 1e-04,
  start_method = "both",
  three = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>
<p>A matrix or dataframe containing gene expression data.
Rows are genes and columns are samples. Must supply one of filename and dat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>Name of file containing gene data. Can be either .csv 
or space separated .dat. Rows are genes and columns are samples. 
Must supply one of filename and dat.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random_starts</code></td>
<td>
<p>The number of random initializations used per gene when
fitting mixtures of t-distributions. Initialization uses k-means by default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_it</code></td>
<td>
<p>The maximum number of iterations per mixture fit. 
Default value is 100.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ll_thresh</code></td>
<td>
<p>The difference in -2 log lambda used as a threshold 
for selecting between g=1 and g=2 for each gene. Default value is 8,
which was chosen arbitrarily in the original paper.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_clust_size</code></td>
<td>
<p>The minimum number of observations per cluster
used when fitting mixtures of t-distributions for each gene.
Default value is 8.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tol</code></td>
<td>
<p>Tolerance value used for detecting convergence of EMMIX fits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start_method</code></td>
<td>
<p>Default value is "both". 
Can also choose "random" for purely random starts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>three</code></td>
<td>
<p>Also test g=2 vs g=3 where appropriate. Defaults to FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An EMMIXgene object containing:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>stat</code></td>
<td>
<p>The difference in log-likelihood for g=1 
and g=2 for each gene (or for g=2 and g=3 where relevant).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>The selected number of components for each gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>it</code></td>
<td>
<p>The number of iterations for each genes selected fit.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selected</code></td>
<td>
<p>An indicator for each genes selected status</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ranks</code></td>
<td>
<p>selected gene ids ranked by stat</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>A dataframe of selected genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all_genes</code></td>
<td>
<p>Returns dat or contents of filename.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">#only run on first 100 genes for speed
alon_sel &lt;- select_genes(alon_data[seq_len(100), ]) 

</code></pre>


</div>