<div class="container">

<table style="width: 100%;"><tr>
<td>Predoss.Feature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Encoding nucleotide sequences using all possible di-nucleotide dependencies.
</h2>

<h3>Description</h3>

<p>In this encoding, not only the adjecent dependencies are considered, but also the association that exists among non-adjacent nucleotides. In MM1, PN.FDTF features, only the dependencies between adjacent nucleotides are taken into account. Though all possible pair-wise dependencies are first introduced by Meher <em>et al</em>. (2014) for predicting splice sites through probablistic approach, the same authors further used this association to encode the splice site dataset for prediction using machine learning classifiers (Meher <em>et al</em>., 2016).  
</p>


<h3>Usage</h3>

<pre><code class="language-R">Predoss.Feature(positive_class, negative_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>negative_class</code></td>
<td>

<p>Sequence dataset of the negative class, must be an object of class <code>DNAStringSet</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be of an object of class <code>DNAStringSet</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This encoding approach will be helpful for transformation of nucleotide sequences into numeric feature vectors, which can subsequently be used as input in several supervised learning models for classification.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*n^{2}</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Note</h3>

<p>Dimension of the feature space will increase geometrically with increase in the length of the sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li>
<p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2014). A statistical approach for 5' splice site prediction using short sequence motifs and without encoding sequence data. <em>BMC Bioinformatics</em>, 15(1), 362.
</p>
</li>
<li>
<p>Meher, P.K., Sahu, T.K., Rao, A.R. and Wahi, S.D. (2016). A computational approach for prediction of donor splice sites with improved accuracy. <em>Journal of Theoretical Biology</em>, 404: 285-294.
</p>
</li>
</ol>
<h3>Examples</h3>

<pre><code class="language-R">data(droso)
positive &lt;- droso$positive
negative &lt;- droso$negative
test &lt;- droso$test
pos &lt;- positive[1:200]
neg &lt;- negative[1:200]
tst &lt;- test
enc &lt;- Predoss.Feature(positive_class=pos, negative_class=neg, test_seq=tst)
enc
</code></pre>


</div>