<div class="container">

<table style="width: 100%;"><tr>
<td>MM2.Feature</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Mapping nucleotide sequences onto numeric feature vectors based on second order nucleotide dependencies.
</h2>

<h3>Description</h3>

<p>This encoding procedure is similar to the MM1 encoding. The only difference is consideration of second order dependencies unlike first order in <code>MM2.Feature</code>. This technique was first conceptualized by Rajapakse and Ho (2005), and adopted by Maji and Garg (2014). The number of parameters to be estimated in MM2 is 64, which is higher than that of MM1 i.e., 16. Further, only the positive class dataset is used for encoding of sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MM2.Feature(positive_class, test_seq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>positive_class</code></td>
<td>

<p>Sequence dataset of the positive class, must be an object of class <code>DNAStringSet</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test_seq</code></td>
<td>

<p>Sequences to be encoded into numeric vectors, must be an object of class <code>DNAStringSet</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For getting an object of class <code>DNAStringSet</code>, the FASTA sequences should be read using the function <em>readDNAStringSet</em> avialble in the <strong>Biostrings</strong> package.
</p>


<h3>Value</h3>

<p>A numeric matrix of order <code class="reqn">m*(n-2)</code>, where <code class="reqn">m</code> is the number of sequences in <code>test_seq</code> and <code class="reqn">n</code> is the length of sequence.
</p>


<h3>Author(s)</h3>

<p>Prabina Kumar Meher, Indian Agricultural Statistics Research Institute, New Delhi-110012, INDIA
</p>


<h3>References</h3>


<ol>
<li> <p>Rajapakse, J. and Ho, L.S. (2005). Markov encoding for detecting signals in genomic sequences. <em>IEEE/ACM Trans Comput Biol Bioinf</em>., 2(2): 131-142.
</p>
</li>
<li> <p>Maji, S. and Garg, D. (2014). Hybrid approach using SVM and MM2 in splice site junction identification. <em>Current Bioinformatics</em>, 9(1): 76-85.
</p>
</li>
</ol>
<h3>See Also</h3>

<p><code>MM1.Feature</code>, <code>WAM.Feature</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(droso)
positive &lt;- droso$positive
test &lt;- droso$test
pos &lt;- positive[1:200]
tst &lt;- test
enc &lt;- MM2.Feature(positive_class=pos, test_seq=tst)
enc
</code></pre>


</div>