<div class="container">

<table style="width: 100%;"><tr>
<td>enve.recplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Enveomics: Recruitment Plots</h2>

<h3>Description</h3>

<p>Produces recruitment plots provided that BlastTab.catsbj.pl has
been previously executed. Requires the <span class="pkg">gplots</span> library.
</p>


<h3>Usage</h3>

<pre><code class="language-R">enve.recplot(
  prefix,
  id.min = NULL,
  id.max = NULL,
  id.binsize = NULL,
  id.splines = 0,
  id.metric = "id",
  id.summary = "sum",
  pos.min = 1,
  pos.max = NULL,
  pos.binsize = 1000,
  pos.splines = 0,
  rec.col1 = "white",
  rec.col2 = "black",
  main = NULL,
  contig.col = grey(0.85),
  ret.recplot = FALSE,
  ret.hist = FALSE,
  ret.mode = FALSE,
  id.cutoff = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>Path to the prefix of the BlastTab.catsbj.pl output files. At
least the files <strong>.rec</strong> and <strong>.lim</strong> must exist with this prefix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.min</code></td>
<td>
<p>Minimum identity to be considered. By default, the minimum detected
identity. This value is a percentage.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.max</code></td>
<td>
<p>Maximum identity to be considered. By default, 100%.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.binsize</code></td>
<td>
<p>Size of the identity bins (vertical histograms). By default, 0.1 for
identity metrics and 5 for bit score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the identity histogram. Zero (0) for
no splines. A generally good value is 1/2. If non-zero, requires the
<span class="pkg">stats</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.metric</code></td>
<td>
<p>Metric of identity to be used (Y-axis). It can be any unambiguous prefix of:
</p>

<ul>
<li>
<p> "identity"
</p>
</li>
<li>
<p> "corrected identity" 
</p>
</li>
<li>
<p> "bit score"
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.summary</code></td>
<td>
<p>Method used to build the identity histogram (Horizontal axis of the right
panel). It can be any unambiguous prefix of: 
</p>

<ul>
<li>
<p> "sum"
</p>
</li>
<li>
<p> "average" 
</p>
</li>
<li>
<p> "median"
</p>
</li>
<li>
<p> "90% lower bound"
</p>
</li>
<li>
<p> "90% upper bound"
</p>
</li>
<li>
<p> "95% lower bound"
</p>
</li>
<li>
<p> "95% upper bound"
</p>
</li>
</ul>
<p>The last four options
correspond to the upper and lower boundaries of the 90% and 95% empirical
confidence intervals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.min</code></td>
<td>
<p>Minimum (leftmost) position in the reference (concatenated) genome (in bp).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.max</code></td>
<td>
<p>Maximum (rightmost) position in the reference (concatenated) genome (in bp).
By default: Length of the genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.binsize</code></td>
<td>
<p>Size of the position bins (horizontal histograms) in bp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos.splines</code></td>
<td>
<p>Smoothing parameter for the splines in the position histogram. Zero (0) for
no splines. If non-zero, requires the stats package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rec.col1</code></td>
<td>
<p>Lightest color in the recruitment plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rec.col2</code></td>
<td>
<p>Darkest color in the recruitment plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>Title of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contig.col</code></td>
<td>
<p>Color of the Contig boundaries. Set to <code>NA</code> to ignore Contig boundaries.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ret.recplot</code></td>
<td>
<p>Indicates if the matrix of the recruitment plot is to be returned.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ret.hist</code></td>
<td>
<p>Ignored, for backwards compatibility.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ret.mode</code></td>
<td>
<p>Indicates if the mode of the identity is to be computed. It requires the 
<span class="pkg">modeest</span> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id.cutoff</code></td>
<td>
<p>Minimum identity to consider an alignment as "top". By default, it is 0.95
for the identity metrics and 95% of the best scoring alignment for bit
score.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Indicates if the function should report the advance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Any additional graphic parameters to be passed to plot for all panels except
the recruitment plot (lower-left).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a list with the following elements:
</p>

<dl>
<dt><code>pos.marks</code></dt>
<dd>
<p>Midpoints of the position histogram.</p>
</dd>
<dt><code>id.matrix</code></dt>
<dd>
<p>Midpoints of the identity histogram.</p>
</dd>
<dt><code>recplot</code></dt>
<dd>
<p>Matrix containing the recruitment plot values 
(if <code>ret.recplot=TRUE</code>).</p>
</dd>
<dt><code>id.mean</code></dt>
<dd>
<p>Mean identity.</p>
</dd>
<dt><code>id.median</code></dt>
<dd>
<p>Median identity.</p>
</dd>
<dt><code>id.mode</code></dt>
<dd>
<p>Mode of the identity (if <code>ret.mode=TRUE</code>).
Deprecated.</p>
</dd>
<dt><code>id.hist</code></dt>
<dd>
<p>Values of the identity histogram
(if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>pos.hist.low</code></dt>
<dd>
<p>Values of the position histogram (depth) with
"low" identity (i.e., below id.cutoff) (if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>pos.hist.top</code></dt>
<dd>
<p>Values of the position histogram (depth) with
"top" identity (i.e., above id.cutoff) (if <code>ret.hist=TRUE</code>).</p>
</dd>
<dt><code>id.max</code></dt>
<dd>
<p>Value of <code>id.max</code>. This is returned because 
<code>id.max=NULL</code> may vary.</p>
</dd>
<dt><code>id.cutoff</code></dt>
<dd>
<p>Value of <code>id.cutoff</code>. 
This is returned because <code>id.cutoff=NULL</code> may vary.</p>
</dd>
<dt><code>seqdepth.mean.top</code></dt>
<dd>
<p>Average sequencing depth with identity above 
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.mean.low</code></dt>
<dd>
<p>Average sequencing depth with identity below 
<code>id.cutoff</code>.</p>
</dd> 
<dt><code>seqdepth.mean.all</code></dt>
<dd>
<p>Average sequencing depth without identity
filtering.</p>
</dd>
<dt><code>seqdepth.median.top</code></dt>
<dd>
<p>Median sequencing depth with identity above
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.median.low</code></dt>
<dd>
<p>Median sequencing depth with identity below
<code>id.cutoff</code>.</p>
</dd>
<dt><code>seqdepth.median.all</code></dt>
<dd>
<p>Median sequencing depth without identity
filtering.</p>
</dd>
<dt><code>id.metric</code></dt>
<dd>
<p>Full name of the used identity metric.</p>
</dd>
<dt><code>id.summary</code></dt>
<dd>
<p>Full name of the summary method used to build the
identity plot.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Luis M. Rodriguez-R [aut, cre]
</p>


</div>