<div class="container">

<table style="width: 100%;"><tr>
<td>env.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build Trees Based on the Environment Around the Indicated Amino Acid(s)</h2>

<h3>Description</h3>

<p>Builds trees based on the environment around the indicated amino acid(s).
</p>


<h3>Usage</h3>

<pre><code class="language-R">env.fasta(file, r = 10, aa = 'all', out.file = 'any')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>path to the single multispecies fasta file to be used as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>a positive integer indicating the radius of the sequence segment considered as environment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aa</code></td>
<td>
<p>the amino acid(s) to be used to encoded the species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.file</code></td>
<td>
<p>path and name of output file. Only if intermediate results data want to be saved (see details).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function builds alignment-independent phylogenetic trees. The input data is a fasta file. When an out.file path is provided, the environment sequences of each species and the vector representing each species are saved in the path provided.
</p>


<h3>Value</h3>

<p>A list with two objects, the first one is an inter-species distance matrix. The second one is an object of class 'phylo'.
</p>


<h3>See Also</h3>

<p>envnj(), fastaconc()
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: env.fasta(file = "./data_t/sample5.fasta")
</code></pre>


</div>