<div class="container">

<table style="width: 100%;"><tr>
<td>envnj</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Build Trees Based on the Environment Around the Indicated Amino Acid(s)</h2>

<h3>Description</h3>

<p>Builds trees based on the environment around the indicated amino acid(s).
</p>


<h3>Usage</h3>

<pre><code class="language-R">envnj(data, r = 10, aa = 'all', metric = "cosine", clustering = "nj", outgroup = 'any')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>input data must be a dataframe where each row corresponds to a protein sequence and each column to a species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r</code></td>
<td>
<p>a positive integer indicating the radius of the sequence segment considered as environment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aa</code></td>
<td>
<p>the amino acid(s) to be used to encoded the species.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric</code></td>
<td>
<p>character string indicating the metric (see metrics() to see the methods allowed).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clustering</code></td>
<td>
<p>string indicating the clustering method, either "nj" or "upgma".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outgroup</code></td>
<td>
<p>when a rooted tree is desired, it indicates the species to be used as outgroup.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function builds alignment-independent phylogenetic trees.
</p>


<h3>Value</h3>

<p>A list with two objects, the first one is an inter-species distance matrix. The second one is an object of class 'phylo'.
</p>


<h3>See Also</h3>

<p>otu.space(), metrics()
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(bovids)
envnj(bovids[, 7:11], aa = "all", outgroup = "Pseudoryx_nghetinhensis")

</code></pre>


</div>