<div class="container">

<table style="width: 100%;"><tr>
<td>ngraMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute n-Gram Frequencies Dataframe</h2>

<h3>Description</h3>

<p>Computes the n-gram frequencies dataframe for the protein and species provides.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ngraMatrix(data, k = 4, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a dataframe with as many columns as species and one row per orthologous protein. The rows and columns must be named accordingly.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>a positive integer, between 1 and 5, indicating the k-mer of the words to be counted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>silent</code></td>
<td>
<p>logical, set to FALSE to avoid loneliness.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The argument prot can be obtained using orth() and orth.seq().
</p>


<h3>Value</h3>

<p>A list with two dataframes. The first one with nsp * npr columns (nsp: number of species, npr: number of proteins per species) and npe rows (npe: number of peptides, 20 for n = 1, 400 for n = 2, 8000 for n = 3 and 160000 for n = 4). The entries of the dataframe are the number of times that the indicated peptide has been counted in the given protein. Orthologous proteins are in consecutive columns, thus the first nsp columns are the orthologous of protein 1 and so on. The second dataframe contains the Species Vector Sums (each vector describes one species).
</p>


<h3>References</h3>

<p>Stuart et al. Bioinformatics 2002; 18:100-108.
</p>


<h3>See Also</h3>

<p>ngram(), svdgram()
</p>


<h3>Examples</h3>

<pre><code class="language-R">ngraMatrix(bovids[,1:3], k = 2)
</code></pre>


</div>