<div class="container">

<table style="width: 100%;"><tr>
<td>getHaplo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve haplotypes.</h2>

<h3>Description</h3>

<p>Retrieve haplotypes from the population.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getHaplo(pop)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>a valid <code>Population</code> object</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Retrieves the haplotypes from both the population which, when
added together, form the genotypes.
</p>


<h3>Value</h3>

<p>A list is returned with two elements, corresponding to the
two haplotype matrices.
</p>


<h3>Author(s)</h3>

<p>Dion Detterer, Paul Kwan, Cedric Gondro
</p>


<h3>See Also</h3>

<p><code>getAddCoefs</code>, <code>getAddOffset</code>, <code>getPhased</code>, <code>getGeno</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Construct a new population with additive effects
pop &lt;- Population(
  popSize = 20, map = map100snp, QTL = 20,
  broadH2 = 0.4, narrowh2 = 0.4, traitVar = 40,
  alleleFrequencies = runif(100, 0.05, 0.5)
)
pop &lt;- addEffects(pop)

# Find the additive contribution to the individuals' phenotypes
hap &lt;- getHaplo(pop)
hap &lt;- (hap[[1]] + hap[[2]])[, getQTL(pop)$Index]
(hap %*% getAddCoefs(pop))[, 1] + getAddOffset(pop)
</code></pre>


</div>