<div class="container">

<table style="width: 100%;"><tr>
<td>epmc_db</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve external database entities referenced in a given publication</h2>

<h3>Description</h3>

<p>This function returns EBI database entities referenced in a publication from
Europe PMC RESTful Web Service.
</p>


<h3>Usage</h3>

<pre><code class="language-R">epmc_db(
  ext_id = NULL,
  data_src = "med",
  db = NULL,
  limit = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ext_id</code></td>
<td>
<p>character, publication identifier</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data_src</code></td>
<td>
<p>character, data source, by default Pubmed/MedLine index will
be searched.
</p>
<p>The following three letter codes represent the sources
Europe PubMed Central supports:
</p>

<dl>
<dt>agr</dt>
<dd>
<p>Agricola is a bibliographic database of citations to the
agricultural literature created by the US National Agricultural Library
and its co-operators.</p>
</dd>
<dt>cba</dt>
<dd>
<p>Chinese Biological Abstracts</p>
</dd>
<dt>ctx</dt>
<dd>
<p>CiteXplore</p>
</dd>
<dt>eth</dt>
<dd>
<p>EthOs Theses, i.e. PhD theses (British Library)</p>
</dd>
<dt>hir</dt>
<dd>
<p>NHS Evidence</p>
</dd>
<dt>med</dt>
<dd>
<p>PubMed/Medline NLM</p>
</dd>
<dt>nbk</dt>
<dd>
<p>Europe PMC Book metadata</p>
</dd>
<dt>pat</dt>
<dd>
<p>Biological Patents</p>
</dd>
<dt>pmc</dt>
<dd>
<p>PubMed Central</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>character, specify database:
</p>

<dl>
<dt>'ARXPR'</dt>
<dd>
<p>Array Express, a database of functional genomics experiments</p>
</dd>
<dt>'CHEBI'</dt>
<dd>
<p>a database and ontology of chemical entities of biological
interest</p>
</dd>
<dt>'CHEMBL'</dt>
<dd>
<p>a database of bioactive drug-like small molecules</p>
</dd>
<dt>'EMBL'</dt>
<dd>
<p>now ENA, provides a comprehensive record of the world's
nucleotide sequencing information</p>
</dd>
<dt>'INTACT'</dt>
<dd>
<p>provides a freely available, open
source database system and analysis tools for molecular interaction data</p>
</dd>
<dt>'INTERPRO'</dt>
<dd>
<p>provides functional analysis of proteins by classifying
them into families and predicting domains and important sites</p>
</dd>
<dt>'OMIM'</dt>
<dd>
<p>a comprehensive and authoritative compendium of human genes and
genetic phenotypes</p>
</dd>
<dt>'PDB'</dt>
<dd>
<p>European resource for the collection,
organisation and dissemination of data on biological macromolecular
structures</p>
</dd>
<dt>'UNIPROT'</dt>
<dd>
<p>comprehensive and freely accessible
resource of protein sequence and functional information</p>
</dd>
<dt>'PRIDE'</dt>
<dd>
<p>PRIDE Archive - proteomics data repository</p>
</dd>
</dl>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>limit</code></td>
<td>
<p>integer, number of results. By default, this function
returns 100 records.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical, print some information on what is going on.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Cross-references as data.frame
</p>


<h3>Examples</h3>

<pre><code class="language-R">  ## Not run: 
  epmc_db("12368864", db = "uniprot", limit = 150)
  epmc_db("25249410", db = "embl")
  epmc_db("14756321", db = "uniprot")
  epmc_db("11805837", db = "pride")
  
## End(Not run)
</code></pre>


</div>