<div class="container">

<table style="width: 100%;"><tr>
<td>SuperMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
creates a supermatrix from multiple gene alignments</h2>

<h3>Description</h3>

<p>combines all alignments in a folder into a single supermatrix</p>


<h3>Usage</h3>

<pre><code class="language-R">SuperMatrix(missing = "-", prefix = "concatenated", save = T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>missing</code></td>
<td>
<p>the character to use when no data is available for a taxa</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p>prefix for the resulting supermatrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save</code></td>
<td>
<p>if True then supermatrix and partition file will be saved</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function reads all fasta format alignments in the working directory and constructs a single supermatrix that includes all taxa present in any of the fasta files and inserts missing symbols for taxa that are missing sequences for some loci.</p>


<h3>Value</h3>

<p>A list with two elements is returned. The first element contains partition data that records the alignment positions of each input fasta file in the combined supermatrix.  The second element is a dataframe version of the supermatrix. If the argument save is set to True then both of these files are also saved to the working directory.
</p>


<h3>Author(s)</h3>

<p>Heath Blackmon
</p>


<h3>References</h3>

<p><a href="http://coleoguy.github.io/">http://coleoguy.github.io/</a></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
SuperMatrix(missing = "N", prefix = "DATASET2", save = T)

## End(Not run)
</code></pre>


</div>